### PR TITLE
fix: add input validation for adding a contact network address

### LINKED ISSUE #257: ️🛠 - Add input validation for adding a network address

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/views/dashboard/contact-book/drawers/AddNetworkAddressDrawer.svelte [hunk 1]
```diff
@@ -4,20 +4,56 @@
...
+    function validate(): boolean {
+        let handledError = false
+        for (const input of [addressInput, addressNameInput, networkSelectionInput]) {
+            try {
+                input.validate()
+            } catch (err) {
+                handledError = true
+            }
+        }
+        return !handledError
```
Comment (MarkNerdi) [inline packages/desktop/views/dashboard/contact-book/drawers/AddNetworkAddressDrawer.svelte]:
```suggestion
    function validate(): boolean {
        for (const input of [addressInput, addressNameInput, networkSelectionInput]) {
            try {
                input.validate()
            } catch (err) {
                return false
            }
        }
        return true
```

File: packages/shared/src/lib/core/contact/utils/validateContactAddressName.ts [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+import { ContactManager } from '../classes'
 import { CONTACT_NAME_MAX_LENGTH } from '../constants'
+import { IContactAddress } from '../interfaces'
+
+export function validateContactAddressName(options: IValidationOptions, contactId?: string, networkId?: string): void {
+    const { isRequired, mustBeUnique, checkLength } = options
 
-export function validateContactAddressName(name: string): void {
-    if (!name) {
+    const name = options?.value as string
```
Comment (MarkNerdi) [inline packages/shared/src/lib/core/contact/utils/validateContactAddressName.ts]: no need for the `?`

File: packages/shared/src/lib/core/contact/utils/validateContactAddressName.ts [hunk 3]
```diff
@@ -15,8 +22,13 @@ export function validateContactAddressName(name: string): void {
         )
     }
 
-    /**
-     * NOTE: We do not need to validate that the name is unique and not being used b/c the user
-     * is adding a contact for the first time when they are in this drawer.
-     */
+    if (mustBeUnique) {
+        const contactAddressMap = ContactManager.getNetworkContactAddressMapForContact(contactId )?.[networkId]
```
Comment (MarkNerdi) [inline packages/shared/src/lib/core/contact/utils/validateContactAddressName.ts]:
```suggestion
        const contactAddressMap = ContactManager.getNetworkContactAddressMapForContact(contactId)?.[networkId]
```

File: packages/shared/src/lib/core/utils/interfaces/validation-options.interface.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export interface IValidationOptions {
```
Comment (MarkNerdi) [inline packages/shared/src/lib/core/utils/interfaces/validation-options.interface.ts:1]: really nice idea about this! i think we can think about using this for each validation funktion

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
looks good!
added some minor comments

File: packages/shared/src/lib/core/utils/interfaces/validation-options.interface.ts (hunk 4 above)
Comment (maxwellmattryan) [reply packages/shared/src/lib/core/utils/interfaces/validation-options.interface.ts:1]: Yeah, Nicole and I came up with the idea during a pair programming session. I think with a validation function available on the input component it makes it a lot easier to write the validation code imo. It doesn't currently support `Promise`s atm but we can fix that in the future when called for :)

**Review by MarkNerdi — APPROVED** [review]
LGTM

**Pull request merged**